### DISCUSSION TITLE
What am I missing?

### DISCUSSION THREAD

**Discussion by ivanji** [jscutlery/semver#546] [General]
I'm using the latest NX version, and attempting to set semver up for the first time. When I go through the installation process I enter an interactive mode where I specify the projects, etc. After this is completed nothing happens (no files are changed). I thought this would immediately create some tasks that we could run later on. If I move to the next step in the readme (Usage) - `nx run my-project:version` nothing happens either, which I believe it's because during the first step there was no target version created.

Am I understanding this correctly?

**Comment by ivanji** [comment]
OK, I've explored this a bit further and realised that it's because we mainly rely on package.json instead of project.json on each of our libraries. project.json only had a root specified, if I add the targets property then when I run semver install, the version target gets added and everything from them on works as expected.

**Reply by edbzn** [reply]
Hi @ivanji, do you mean that you had no workspace.json configuration?

**Reply by ivanji** [reply]
I do have a workspace.json configuration, but nowadays NX only uses it to specify the projects locations. The problem I encountered initially was because we were not using project.json files within any of our NX projects. Just relying on package.json - which I believe it's a valid approach - based on this https://nx.dev/configuration/packagejson. At the same time if we wanted to make use of NX plugins we'd need to have one for each package (with the correct schema).

**Reply by edbzn** [reply]
Ok, yeah actually we don't handle workspace without global `workspace.json` configuration, I opened an issue to fix that, but I think there's another issue with `package.json` projects only as well.

**Reply by jpetri-de** [reply]
@edbzn, I ran into the same problem: I do not have a workspace.json in a NX 14 workspace. Do you have a solution or an idea for a workaround? Thanks a lot!
